diff --git a/tensorflow/lite/core/interpreter_builder.cc b/tensorflow/lite/core/interpreter_builder.cc
index 4623c0bf..3cbfe9ce 100644
--- a/tensorflow/lite/core/interpreter_builder.cc
+++ b/tensorflow/lite/core/interpreter_builder.cc
@@ -75,6 +75,8 @@ limitations under the License.
 #endif
 #endif  // defined(__APPLE__)
 
+#define RTK_REMOVE_EXTRA_QUANTIZATION_DATA (1)
+
 namespace tflite {
 
 namespace {
@@ -419,9 +421,13 @@ TfLiteStatus InterpreterBuilder::ParseQuantization(
     return kTfLiteError;
   }
 
+  int quantization_data_is_removed = 0;
   // Ensure that the number of scales matches the number of zero_points.
   if (src_quantization->scale()->size() !=
       src_quantization->zero_point()->size()) {
+#ifdef RTK_REMOVE_EXTRA_QUANTIZATION_DATA
+    quantization_data_is_removed = 1;
+#else
     TF_LITE_REPORT_ERROR(
         error_reporter_,
         "QuantizationParam has %d zero_point values and %d scale values. "
@@ -430,6 +436,7 @@ TfLiteStatus InterpreterBuilder::ParseQuantization(
         src_quantization->zero_point()->size(),
         src_quantization->scale()->size());
     return kTfLiteError;
+#endif
   }
 
   const size_t num_scales = src_quantization->scale()->size();
@@ -466,8 +473,13 @@ TfLiteStatus InterpreterBuilder::ParseQuantization(
   affine_quantization->zero_point = TfLiteIntArrayCreate(num_scales);
   for (size_t i = 0; i < num_scales; ++i) {
     affine_quantization->scale->data[i] = src_quantization->scale()->Get(i);
-    affine_quantization->zero_point->data[i] =
+    if (quantization_data_is_removed) {
+      affine_quantization->zero_point->data[i] =
+        src_quantization->zero_point()->Get(0);
+    } else {
+          affine_quantization->zero_point->data[i] =
         src_quantization->zero_point()->Get(i);
+    }
   }
   affine_quantization->quantized_dimension =
       src_quantization->quantized_dimension();
