diff --git a/tensorflow/lite/kernels/conv.cc b/tensorflow/lite/kernels/conv.cc
index befd60da71b..a1224e52672 100644
--- a/tensorflow/lite/kernels/conv.cc
+++ b/tensorflow/lite/kernels/conv.cc
@@ -772,9 +772,9 @@ void EvalQuantizedPerChannel(TfLiteContext* context, TfLiteNode* node,
 
   const int8_t* filter_data;
   const size_t bytes_unpacked = filter->bytes * 2;
-  auto unpacked_filter_data = std::make_unique<int8_t[]>(bytes_unpacked);
-
+  std::unique_ptr<int8_t[]> unpacked_filter_data = nullptr;
   if (filter->type == kTfLiteInt4) {
+    unpacked_filter_data = std::make_unique<int8_t[]>(bytes_unpacked);
     tflite::tensor_utils::UnpackDenseInt4IntoInt8(
         GetTensorData<int8_t>(filter), GetTensorShape(filter).FlatSize(),
         unpacked_filter_data.get());
